apps/condo/domains/acquiring/tasks/utils/queries.js

```diff
@@ -0,0 +1,412 @@
+const Big = require('big.js')
+const dayjs = require('dayjs')
+const { get, isNil, isUndefined } = require('lodash')
+
+const { getLogger } = require('@open-condo/keystone/logging')
+
+const {
+    PAYMENT_DONE_STATUS,
+    PAYMENT_WITHDRAWN_STATUS,
+} = require('@condo/domains/acquiring/constants/payment')
+const {
+    RECURRENT_PAYMENT_INIT_STATUS,
+    RECURRENT_PAYMENT_DONE_STATUS,
+    RECURRENT_PAYMENT_ERROR_NEED_RETRY_STATUS,
+    RECURRENT_PAYMENT_ERROR_STATUS,
+} = require('@condo/domains/acquiring/constants/recurrentPayment')
+const {
+    dvAndSender,
+    PAYMENT_ERROR_UNKNOWN_CODE,
+    PAYMENT_ERROR_LIMIT_EXCEEDED_CODE,
+    PAYMENT_ERROR_CONTEXT_NOT_FOUND_CODE,
+    PAYMENT_ERROR_CONTEXT_DISABLED_CODE,
+    PAYMENT_ERROR_CARD_TOKEN_NOT_VALID_CODE,
+    PAYMENT_ERROR_CAN_NOT_REGISTER_MULTI_PAYMENT_CODE,
+} = require('@condo/domains/acquiring/tasks/utils/constants')
+const {
+    RecurrentPayment,
+    RecurrentPaymentContext,
+    Payment,
+    registerMultiPayment: registerMultiPaymentMutation,
+    MultiPayment,
+} = require('@condo/domains/acquiring/utils/serverSchema')
+const {
+    BillingReceipt,
+} = require('@condo/domains/billing/utils/serverSchema')
+const {
+    RECURRENT_PAYMENT_PROCEEDING_SUCCESS_RESULT_MESSAGE_TYPE,
+    RECURRENT_PAYMENT_PROCEEDING_FAILURE_RESULT_MESSAGE_TYPE,
+    RECURRENT_PAYMENT_TOMORROW_PAYMENT_MESSAGE_TYPE,
+} = require('@condo/domains/notification/constants/constants')
+const { sendMessage } = require('@condo/domains/notification/utils/serverSchema')
+const {
+    ServiceConsumer,
+} = require('@condo/domains/resident/utils/serverSchema')
+
+const logger = getLogger('recurrent-payment-processing-queries')
+
+async function getAllReadyToPayRecurrentPaymentContexts (context, date, pageSize, offset, extraArgs = {}) {
```

Comment:
it looks like serverSideUtil

Reply:
Query inside getAllReadyToPayRecurrentPaymentContexts has too much business logic. Not sure if we should move BL specific queries into the serverSide utils package